src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1532,23 +1536,97 @@ def get_windows_admin_username_and_password(
 
         return self._get_windows_admin_username_and_password()
 
-    def get_enable_ahub(self) -> bool:
-        """Obtain the value of enable_ahub.
+    def get_windows_admin_password(self) -> Union[str, None]:
+        """Obtain the value of windows_admin_password.
+
+        Note: This function is intended to be used in update mode.
+
+        :return: string or None
+        """
+        # read the original value passed by the command
+        windows_admin_password = self.raw_param.get("windows_admin_password")
+
+        # this parameter does not need dynamic completion
+        # this parameter does not need validation
+        return windows_admin_password
+
+    # pylint: disable=unused-argument
+    def _get_enable_ahub(
+        self, enable_validation: bool = False, **kwargs
+    ) -> bool:
+        """Internal function to obtain the value of enable_ahub.
 
         Note: enable_ahub will not be directly decorated into the `mc` object.
 
+        This function supports the option of enable_validation. When enabled, if both enable_ahub and disable_ahub are
+        specified, raise a MutuallyExclusiveArgumentError.
+
         :return: bool
         """
         # read the original value passed by the command
         enable_ahub = self.raw_param.get("enable_ahub")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if self.mc and self.mc.windows_profile:
-            enable_ahub = self.mc.windows_profile.license_type == "Windows_Server"
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object
```

Comment:
What does it mean `In create mode`? enable/disable ahub can be used in both create and update. But to change `Windows_Server`, it should be an update operation

Reply:
Currently, we are refactoring `aks create` and `aks update`, and `In create mode` means this only applies to the case of `aks create` but not `aks update`.

I suppose we have the `--enable-ahub` option for both `aks create` and `aks update`, but for option `--disable-ahub`, it is only supported by `aks update`.

Reply:
Back to the topic, in the create mode (`aks create`), when determining the value of a parameter, obtaining from `mc` takes precedence over user's input. While in the update mode (`aks update`), we do not obtain the value from `mc`.

Reply:
Since we do not obtain the value from mc when in the update mode, why do we not use the same for the create model?

Reply:
In the update mode, the parameter value is not obtained from `mc` because we just need to use the value provided by the user to overwrite the value in `mc`.

But in the create mode, `mc` is completely blank, and we want to assemble the user's input into a complete `mc`. Considering that our parameters have a certain dependency or mutual exclusion relationship, or the parameters will be normalized before putting into `mc`, there are even some parameters that do not need user to provide explicitly, but can be dynamically completed by cli. In order to ensure the consistency of the state and make the value of the parameter more clearly determined, we created the following principle in the create mode, that is, if a parameter has been placed in the `mc` object, then when trying to get the parameter value again, (e.g. cross-validation with other parameters) the value in `mc` shall prevail.

Going back to the example in the windows profile, the current windows related parameters will not be used after being set into `mc`, but in order to keep the form consistent with other parameters, we keep such acquisition method in the create mode.